src/db_impl.cc

```diff
@@ -837,6 +837,46 @@ Status TitanDBImpl::DeleteFilesInRanges(ColumnFamilyHandle* column_family,
   return s;
 }
 
+int TitanDBImpl::CountSortedRuns(
+    const std::vector<std::shared_ptr<BlobFileMeta>>& files) {
+  std::list<BlobFileMeta*> remained_files;
+  std::vector<int> sorted_run_blobs;
+  int num_files = files.size();
+
+  auto blob_cmp = [](const BlobFileMeta* f1, const BlobFileMeta* f2) {
+    return f1->smallest_key().compare(f2->smallest_key()) < 0;
+  };
+  for (const auto& file : files) {
+    remained_files.emplace_back(file.get());
+  }
+  remained_files.sort(blob_cmp);
+
+  while (!remained_files.empty()) {
+    int cnt = 0;
+    BlobFileMeta* pre = nullptr;
+    for (auto iter = remained_files.begin(); iter != remained_files.end();) {
+      if (pre == nullptr ||
+          (*iter)->smallest_key().compare(pre->largest_key()) > 0) {
+        cnt++;
+        pre = *iter;
+        remained_files.erase(iter++);
+      } else {
+        iter++;
+      }
+    }
+    sorted_run_blobs.push_back(cnt);
+  }
+
+  // There might be several sorted runs consist of few wide range blob files.
+  // Ignore these sorted runs since it has almost no impact to scan performance.
```

Comment:
I don't quite get this optimization, and don't  know to reason about the 10% number. How about make the implementation simpler and skip this optimization?

Reply:
Okay, we can skip this. This optimization is from my previous experiment, that some sorted runs consist of only 10% blob files with this estimate method.